lib/handlers.py

```diff
@@ -25,15 +25,26 @@
     'StatusHandler',
 ]
 
+
 def _is_view_supported(view):
     return view.file_name() is not None and any(
         view.file_name().endswith(ext) for ext in languages.SUPPORTED_EXTS_TO_LANG
     )
 
 
 def _check_view_size(view):
-    # max file size is 75 kilobytes (75 * 1024)
-    return view.size() <= 76800
+    logger.debug('fetching max file size')
+    max_file_size = 1048576  # 1 MB default
+
+    try:
+        resp, body = requests.kited_get('/clientapi/settings/max_file_size_kb')
```

Comment:
I think a better way to do this is to:

1. Keep a global variable `MAX_FILE_SIZE` that stores the max value. Initialize this when the plugin activates. If it is not set, then default to 1 MB.
2. Within `_check_view_size`, first check against the cached `MAX_FILE_SIZE` value. If the file size is under, then everything is ok.
3. If the file size is over (which should be rare), then update `MAX_FILE_SIZE` by pinging the Kite API. Then return whether or not the file size is under the new `MAX_FILE_SIZE` value.

This approach has the benefit of only making the API request once on startup, since it should be rare for users to be working with very large files. `_check_view_size` is called on every select/edit event, so the current implementation could cause some noticeable lag.

Reply:
While I agree that it's a better optimized approach, I'm not sure if it'll make a meaningful difference. I'm using VSCode with this change, and each call to `max_file_size` returns in 30-60 microseconds.

In addition, if we cached the value, it would mean users who reduce their max file size limit would potentially have to restart Sublime to have their editor reflect the new value.

Reply:
Actually now that I think about it, a better strategy may be to just update `MAX_FILE_SIZE` every time a new view is activated via `on_activated`, similar to how the `StatusHandler` initializes the status icon (https://github.com/kiteco/KiteSublime/blob/master/lib/handlers.py#L1085).

Reply:
> While I agree that it's a better optimized approach, I'm not sure if it'll make a meaningful difference. I'm using VSCode with this change, and each call to max_file_size returns in 30-60 microseconds.

This would need to be tested across different hardware specs under different load conditions etc for me to feel comfortable that users likely won't see any degradation when e.g. working with large files, typing very quickly, etc.

Also a Sublime specific concern — The way `kited_get` works is that it uses a pool of 4 connections that's shared across all the event handlers in the plugin, so adding this on every event would increase contention significantly.

Also, personally I'm not a fan of clogging up logs with meaningless lines, though that's more of a personal preference. The first 2 points are what would drive my decision.

Reply:
> This would need to be tested across different hardware specs under different load conditions etc for me to feel comfortable that users likely won't see any degradation when e.g. working with large files, typing very quickly, etc.
 
That's fair. The new strategy you proposed sounds good to me as well.